### ISSUE TITLE
Option to Snap center stroke Edge with pixel grid

### ISSUE THREAD

**Issue by Yakehira** [uxbox/uxbox#2043]
**Is your feature request related to a problem? Please describe.**
A center stroked Path is hard to align with snapping enabled as it tries to snap the path, not the 'real' edge, especially zoomed out it never aligns cause it jumps whole pixels so it's always off-pixel. Even zoomed in it's not easy cause even a 1 pixel center line will jump whole pixels without going into edit mode on it and moving the vector.

**Describe the solution you'd like**
When I add a Center Stroke with 1 or 1.5 (or other uneven sizes) i want the ability to Snap the edge to the pixel grid for better rendering and to work faster, rather than deal with nudging and manual edits on the sidebar, make this experience better please :)

![image](https://user-images.githubusercontent.com/773039/175794284-f42bc016-e9b1-4f50-b158-672141f5896f.png)

**Comment by Alotor** [comment]
Hi @Yakehira , thanks for all the feedback you're giving. We really appreciate it.

We don't see the need for this right now. You can create an "inside" stroke to have this behaviour. 

![image](https://user-images.githubusercontent.com/3418223/176369294-847ba786-6409-429a-892e-04a75f4fbde6.png)

**Comment by Yakehira** [comment]
The rounding behavior is different depending on inside/outside/center, so this is not actually an alternative as you suggest this is why i opened this Issue.

Inside/Outside/Center Example to illustrate the rounding of inside corners.
![image](https://user-images.githubusercontent.com/773039/176433282-64eb1084-e4fa-4460-827e-e96b32b81de2.png)

**Comment by Alotor** [comment]
Umm, I see. You're right.

One feature that is in our backlog that maybe will help you is "Stroke to path" that will make the stroke to be part of the shape. 

Do you think that would help you? If it would, I can link this issue so we can let you know when its developed.

**Comment by Alotor** [comment]
https://tree.taiga.io/project/penpot/us/2956?backlog-q=stroke%20to%20path&no-milestone=1

**Comment by Yakehira** [comment]
I don't think that would satisfy the issue for alignment improvements beyond what Figma offers, the ability to snap the Stroke edges would make it much easier to do pixel-perfect precision work rather than depend on manual number edits/nudge/zoom for precise vector snapping (Which is inaccurate in PenPot currently see https://github.com/penpot/penpot/issues/2040)

**Comment by Yakehira** [comment]
What i originally was asking for seems to have been fixed at some point, so everything now snaps to full and midpoints correctly, allowing for better control.
